data-prepper-plugin-schema/src/test/java/org/opensearch/dataprepper/schemas/PluginConfigsJsonSchemaConverterTest.java

```diff
@@ -0,0 +1,108 @@
+
+    @Mock
+    private Reflections reflections;
+
+    @InjectMocks
+    private PluginConfigsJsonSchemaConverter objectUnderTest;
+
+    @Test
+    void testValidPluginTypeNames() {
+        assertThat(PLUGIN_TYPE_NAME_TO_CLASS_MAP.keySet().containsAll(objectUnderTest.validPluginTypeNames()),
+                is(true));
+    }
+
+    @Test
+    void testPluginTypeNameToPluginTypeWithValidInput() {
+        objectUnderTest.validPluginTypeNames().forEach(
+                pluginType -> assertThat(objectUnderTest.pluginTypeNameToPluginType(pluginType),
+                        equalTo(PLUGIN_TYPE_NAME_TO_CLASS_MAP.get(pluginType))));
+    }
+
+    @Test
+    void testPluginTypeNameToPluginTypeWithInValidInput() {
+        final String inValidPluginType = "invalid-" + UUID.randomUUID();
+        assertThrows(
+                IllegalArgumentException.class, () -> objectUnderTest.pluginTypeNameToPluginType(inValidPluginType));
+    }
+
+    @Test
+    void testConvertPluginConfigsIntoJsonSchemasHappyPath() throws JsonProcessingException {
+        when(reflections.getTypesAnnotatedWith(eq(DataPrepperPlugin.class))).thenReturn(Set.of(TestPlugin.class));
+        final ObjectNode objectNode = OBJECT_MAPPER.createObjectNode();
+        when(jsonSchemaConverter.convertIntoJsonSchema(
+                any(SchemaVersion.class), any(OptionPreset.class), eq(TestPluginConfig.class))).thenReturn(objectNode);
+        final Map<String, String> result = objectUnderTest.convertPluginConfigsIntoJsonSchemas(
+                SchemaVersion.DRAFT_2020_12, OptionPreset.PLAIN_JSON, TestPluginType.class);
+        assertThat(result.size(), equalTo(1));
+        final Map<String, Object> schemaMap = OBJECT_MAPPER.readValue(result.get("test_plugin"), MAP_TYPE_REFERENCE);
+        assertThat(schemaMap, notNullValue());
+        assertThat(schemaMap.containsKey(DOCUMENTATION_LINK_KEY), is(true));
+        assertThat(schemaMap.containsKey(PLUGIN_NAME_KEY), is(true));
+    }
+
+    @Test
+    void testConvertPluginConfigsIntoJsonSchemasWithError() throws JsonProcessingException {
+        when(reflections.getTypesAnnotatedWith(eq(DataPrepperPlugin.class))).thenReturn(Set.of(TestPlugin.class));
+        final JsonProcessingException jsonProcessingException = mock(JsonProcessingException.class);
+        when(jsonSchemaConverter.convertIntoJsonSchema(
+                any(SchemaVersion.class), any(OptionPreset.class), eq(TestPluginConfig.class))).thenThrow(
+                        jsonProcessingException);
+        final Map<String, String> result = objectUnderTest.convertPluginConfigsIntoJsonSchemas(
+                SchemaVersion.DRAFT_2020_12, OptionPreset.PLAIN_JSON, TestPluginType.class);
+        assertThat(result.isEmpty(), is(true));
+    }
+
+    @DataPrepperPlugin(
+            name = "test_plugin", pluginType = TestPluginType.class, pluginConfigurationType = TestPluginConfig.class)
+    static class TestPlugin {
+
+    }
+
+    static class TestPluginConfig {
```

Comment:
We should test with fields in here. Maybe some with and some without the `JsonProperty` annotation.